### ISSUE TITLE
getting go#config#DiagnosticsLevel on load of go project folder

### ISSUE THREAD

**Issue by amitavaghosh1** [fatih/vim-go#3280]
### What did you do? (required: The issue will be **closed** when not provided)
- I open a golang project in neovim as such `nvim .`, 
- I open a golang file
- somtimes as I am typing, when autocomplete kicks in. I get this error message.
I couldn't figure out what is doing this.

### What did you expect to happen?
Not for this error to come.

### What happened instead?
vim-go: initialized gopls
vim-go: Finished loading packages.
vim-go: Finished loading packages.
Error detected while processing function <SNR>159_on_stdout[1]..<SNR>159_neocb[58]..307[10]..310[3]..311[2]..313:
line    1:
E117: Unknown function: go#config#DiagnosticsLevel
Error detected while processing function <SNR>159_on_stdout[1]..<SNR>159_neocb[58]..307[10]..310[3]..311[2]..313:
line    1:
E15: Invalid expression: go#config#DiagnosticsLevel()

####  `vimrc` you used to reproduce:

<details><summary>vimrc</summary>

```vim
Plug 'scrooloose/nerdtree'
Plug 'jistr/vim-nerdtree-tabs'
Plug 'tpope/vim-commentary'
Plug 'tpope/vim-fugitive'
Plug 'vim-airline/vim-airline'
Plug 'vim-airline/vim-airline-themes'
Plug 'airblade/vim-gitgutter'
Plug 'vim-scripts/grep.vim'
Plug 'jiangmiao/auto-pairs'
Plug 'majutsushi/tagbar'
Plug 'w0rp/ale'
Plug 'Yggdroot/indentLine'
Plug 'sheerun/vim-polyglot'
Plug 'tpope/vim-rhubarb' " required by fugitive to :Gbrowse
Plug 'wsdjeg/FlyGrep.vim'
Plug 'ctrlpvim/ctrlp.vim'
Plug 'junegunn/goyo.vim'
Plug 'dag/vim-fish', {'for': ['fish']}
Plug 'tpope/vim-surround'
Plug 'junegunn/fzf', { 'dir': '~/.fzf', 'do': './install --all' }
Plug 'junegunn/fzf.vim'
Plug 'wookayin/fzf-ripgrep.vim'
Plug 'ctrlpvim/ctrlp.vim'


"" Autocomplete
Plug 'neoclide/coc.nvim', {'branch': 'release'}
"" golang

Plug 'fatih/vim-go'

" go
" vim-go
" run :GoBuild or :GoTestCompile based on the go file
function! s:build_go_files()
  let l:file = expand('%')
  if l:file =~# '^\f\+_test\.go$'
    call go#test#Test(0, 1)
  elseif l:file =~# '^\f\+\.go$'
    call go#cmd#Build(0)
  endif
endfunction

let g:go_list_type = "quickfix"
let g:go_fmt_command = "goimports"
let g:go_fmt_fail_silently = 1

let g:go_def_mode='gopls'
let g:go_info_mode='gopls'

let g:go_def_mapping_enabled = 0
let g:go_highlight_types = 1
let g:go_highlight_fields = 1
let g:go_highlight_functions = 1
let g:go_highlight_methods = 1
let g:go_highlight_operators = 1
let g:go_highlight_build_constraints = 1
let g:go_highlight_structs = 1
let g:go_highlight_generate_tags = 1
let g:go_highlight_space_tab_error = 0
let g:go_highlight_array_whitespace_error = 0
let g:go_highlight_trailing_whitespace_error = 0
let g:go_highlight_extra_types = 1

autocmd BufNewFile,BufRead *.go setlocal noexpandtab tabstop=2 shiftwidth=2 softtabstop=2

augroup completion_preview_close
  autocmd!
  if v:version > 703 || v:version == 703 && has('patch598')
    autocmd CompleteDone * if !&previewwindow && &completeopt =~ 'preview' | silent! pclose | endif
  endif
augroup END

augroup go

  au!
  au Filetype go command! -bang A call go#alternate#Switch(<bang>0, 'edit')
  au Filetype go command! -bang AV call go#alternate#Switch(<bang>0, 'vsplit')
  au Filetype go command! -bang AS call go#alternate#Switch(<bang>0, 'split')
  au Filetype go command! -bang AT call go#alternate#Switch(<bang>0, 'tabe')

  au FileType go nmap <Leader>dd <Plug>(go-def-vertical)
  au FileType go nmap <Leader>dv <Plug>(go-doc-vertical)
  au FileType go nmap <Leader>db <Plug>(go-doc-browser)

  au FileType go nmap <leader>r  <Plug>(go-run)
  au FileType go nmap <leader>t  <Plug>(go-test)
  au FileType go nmap <Leader>gt <Plug>(go-coverage-toggle)
  au FileType go nmap <Leader>i <Plug>(go-info)
  au FileType go nmap <silent> <Leader>l <Plug>(go-metalinter)
  au FileType go nmap <C-g> :GoDecls<cr>
  au FileType go nmap <leader>dr :GoDeclsDir<cr>
  au FileType go imap <C-g> <esc>:<C-u>GoDecls<cr>
  au FileType go imap <leader>dr <esc>:<C-u>GoDeclsDir<cr>
  au FileType go nmap <leader>rb :<C-u>call <SID>build_go_files()<CR>
  au FileType go nmap ga :GoImplements<cr>
  au filetype go inoremap <buffer> . .<C-x><C-o>

  nmap <silent> gd <Plug>(coc-definition)
  nmap <silent> gy <Plug>(coc-type-definition)
  nmap <silent> gi <Plug>(coc-implementation)
  nmap <silent> gr <Plug>(coc-references)

augroup END

nmap <leader>rn <Plug>(coc-rename)

" ale
:call extend(g:ale_linters, {
    \"go": ['golint', 'go vet'], })

```
</details>

*if you need more details on vimrc. it's in my [dotfiles](https://github.com/ikouchiha47/dotfiles/blob/master/.config/nvim/init.vim) 

I have run `:GoUpdateBinaries` prior to this.

#### Vim version (first three lines from `:version`):

NVIM v0.4.4
Build type: Release
LuaJIT 2.0.5

#### Go version (`go version`):

go version go1.15.2 darwin/amd64

#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>

GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/amitava.ghosh/Library/Caches/go-build"
GOENV="/Users/amitava.ghosh/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOINSECURE=""
GOMODCACHE="/Users/amitava.ghosh/dev/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/amitava.ghosh/dev/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="off"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/amitava.ghosh/dev/go/src/platform-backend/packages/client-service/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/lq/6mv9lzs94fq1dhg3__j2td5w0000gr/T/go-build837758273=/tmp/go-build -gno-record-gcc-switches -fno-common"

</pre></details>

#### gopls version
<details><summary><code>gopls version</code> Output:</summary><br><pre>

golang.org/x/tools/gopls v0.7.1
    golang.org/x/tools/gopls@v0.7.1 h1:Mh3Z8Xcoq3Zy7ksSlwDV/nzQSbjFf06A+L+F8YHq55U=

</pre></details>

#### vim-go configuration:
<details><summary>vim-go configuration</summary><br><pre>
g:go_fmt_fail_silently = 1
g:go_list_type = 'quickfix'
g:go_highlight_extra_types = 1
g:go_loaded_install = 1
g:go_highlight_operators = 1
g:go_highlight_fields = 1
g:go_highlight_trailing_whitespace_error = 0
g:go_def_mode = 'gopls'
g:go_highlight_structs = 1
g:go_highlight_functions = 1
g:go_gotags_bin = 'gotags'
g:go_highlight_space_tab_error = 0
g:go_highlight_types = 1
g:go_fmt_command = 'goimports'
g:go_jump_to_error = 1
g:go_highlight_methods = 1
g:go_info_mode = 'gopls'
g:go_loaded_gosnippets = 1
g:go_highlight_generate_tags = 1
g:go_highlight_build_constraints = 1
g:go_highlight_array_whitespace_error = 0
g:go_def_mapping_enabled = 0
</pre></details>

#### filetype detection configuration:
<details><summary>filetype detection</summary><br><pre>
filetype detection:ON  plugin:ON  indent:ON
</pre></details>

**Comment by amitavaghosh1** [comment]
This might have to do something with coc , with LanguageClient-neovim and Ultisnips and deoplete its working without issues.

Closing this for now.

**Comment by amitavaghosh1** [comment]
@bhcleek  thank you pointing out. I was also seeing this issue popping up when I was using :Rg to search and also when I was using <C-P> like this `"cnoremap <C-P> <C-R>=expand("%:p:h") . "/" <CR>` and also as soon as I toggled nerdtree on and off.
I don't think coc is the issue. Although I replaced it with LanguageClient-neovim and deoplete with UltiSnips for now. 

And I have moved away from NERDTree to the default netrw with some tweaks to make it look like NERD although I miss it. 
And even now if I turn of the cnoremap mapping there is a different error like channel id is not found (I don't remember exactly)
